taxi/tests/test_view.py

```diff
@@ -0,0 +1,128 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+
+from taxi.models import Manufacturer, Driver, Car
+
+INDEX_URL = reverse("taxi:index")
+MANUFACTURERS_URL = reverse("taxi:manufacturer-list")
+MANUFACTURERS_SEARCH_URL = reverse("taxi:manufacturer-list") + "?name=a"
+DRIVERS_URL = reverse("taxi:driver-list")
+DRIVERS_SEARCH_URL = reverse("taxi:driver-list") + "?username=a"
+CAR_URL = reverse("taxi:car-list")
+CAR_SEARCH_URL = reverse("taxi:car-list") + "?model=a"
+
+
+class PublicIndexTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(INDEX_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PublicManufacturerTest(TestCase):
+    def test_login_required(self):
+        res = self.client.get(MANUFACTURERS_URL)
+        self.assertNotEqual(res.status_code, 200)

```

Comment:
no need to create separate classess for public test, u can create PublicPagesTest with all public tests inside